src/main/java/net/ossindex/gradle/OssIndexPlugin.java

```diff
@@ -61,11 +83,24 @@ private Proxy getProxy(Project project, String scheme) {
     }
 
     private void doAudit(Task task) {
+
+        if (this.settings == null) {
+            this.settings = getAuditExtensions(task.getProject());
+        }
+
+        if (this.junitReport == null) {
+            this.junitReport = settings.junitReport;
+        }
+
         ArtifactGatherer gatherer = factory.getGatherer();
         Set<GradleArtifact> gradleArtifacts = gatherer != null ? gatherer.gatherResolvedArtifacts(task.getProject()) : null;
         DependencyAuditor auditor = factory.getDependencyAuditor(gradleArtifacts, proxies);
 
-        AuditResultReporter reporter = new AuditResultReporter(gradleArtifacts, getAuditExtensions(task.getProject()));
+        reporter = new AuditResultReporter(gradleArtifacts,
+            getAuditExtensions(task.getProject()),
+            this,
+            junitXmlReportWriter,
+            project.getDisplayName().split(" ")[1].replaceAll("\'","") + ":audit");

```

Comment:
I don't know anything about the display name. I'll have to muck about. Is there a chance that the "display name" can have unexpected spaces that would throw off he parsing here?

Reply:
Printing it out the display name, after treatment, is the same as the task name:

> Task :digital:identity-verification-service:audit 
Display Name - project ':digital:identity-verification-service'

This then ends up in the report like so:

<testcase id="46" name=":common:services-common:audit - 0 unignored (of 1 total) vulnerabilities found"/>

It was the first method I found to let me add in the task name to the report. If you know of something simpler or more reliable then let me know and I'll use that instead.